common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -1874,6 +1874,14 @@ def _update_item_from_fields(
                 block_data.fields = settings
 
                 new_id = new_structure['_id']
+
+                # When a draft block from a published structure is reverted in revert_to_published(),
+                # the update_version is updated while source_version is not. If source_version is not None,
+                # set it to None - so that has_changes() will use update_version for comparison and return True.
+                # Prevents the block from getting stuck in the published state.
+                if block_data.edit_info.source_version:
+                    block_data.edit_info.source_version = None
```

Comment:
```source_version``` tells where copied/published blocks cames from just like in case of ```revert_to_published()``` (copy draft from public) and ```publish()``` (copy publish from draft) method the ```source_version``` is set where its copies from so we dont need to update it all time except this 2 cases, while ```update_version``` the guid for the structure where this xblock got its current field values and its updates whenever there is change in block

This is correct behavior for ```source_version``` because whenever the block is copied from other structure the source and destination block are equal by comparison its source_version at ```has_changes()```